File: Labs/Lab09/HistoricoPedidos.java
import java.util.ArrayList;

public class HistoricoPedidos<T> {
    private ArrayList<T> historico;

    public HistoricoPedidos() {
        this.historico = new ArrayList<T>();
    }

    public ArrayList<T> getHistorico() {
        return historico;
    }

    public void setHistorico(ArrayList<T> historico) {
        this.historico = historico;
    }

    public void adicionarAoHistorico(T pedido) {
        historico.add(pedido);
    }

    public void exibirHistorico() {
        for (T pedido : historico) {
            System.out.println(pedido);
        }
    }

}

File: Lab07/HistoricoPedidos.cpp
#include "HistoricoPedidos.hpp"
#include "NodoPedido.hpp"
#include "Pedido.hpp"
#include "Pessoa.hpp"
#include <iostream>
using namespace std;

HistoricoPedidos::HistoricoPedidos() : inicio(nullptr) {}

void HistoricoPedidos::adicionarAoHistorico(Pedido *pedido)
{
    NodoPedido *novoNodo = new NodoPedido(pedido);
    novoNodo->prox = inicio;
    inicio = novoNodo;
    cout << "Pedido concluido: " << pedido->getDescricaoPedido() << "\n";
}

void HistoricoPedidos::exibirHistorico(ListaClientes *listaClientes)
{
    NodoPedido *atual = inicio;
    while (atual != nullptr)
    {
        Pedido *pedido = atual->pedido;
        Pessoa *cliente = listaClientes->buscarCliente(pedido->getCpf());
        if (cliente != nullptr)
        {
            cout << "Codigo: " << pedido->getCodigoPedido()
                 << ", Descricao: " << pedido->getDescricaoPedido()
                 << ", Cliente: " << cliente->nome
                 << " CPF: " << cliente->cpf << "\n";
        }
        atual = atual->prox;
    }
}
